src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1418,7 +1417,22 @@ public void Complete(Exception ex)
                 }
             }
 
-            Output.Complete();
+            // Finalize headers
+            if (!HasResponseStarted)
+            {
+                await FireOnStarting();
+            }
+
+            // Flush headers, body, trailers...
+            if (!HasResponseCompleted)
+            {
+                if (!VerifyResponseContentLength(out var lengthException))
+                {
+                    throw lengthException;
```

Comment:
IS this the code we had before? This should be `ExceptionDispatchInfo.Throw(lengthException)`

Reply:
VerifyResponseContentLength is a factory that creates an exception, it hasn't been thrown before. On other code paths the exception is only logged.